AssertionError: '' acceptable suffix is ['.pt'] (Before running train.py)

## ❔Question
When I was editing configurations in configurations window, I wrote --weights '' in Parameters column to train a  model without pre-training. Then, PyCharm returned me an AssertionError: '' acceptable suffix is ['.pt']. But, when I changed the default value of --weights from 'yolov5s.pt' to '', train.py worked successfully. What is the reason?
## Additional context

--- a/utils/general.py
+++ b/utils/general.py
@@ -298,8 +298,9 @@ def check_suffix(file='yolov5s.pt', suffix=('.pt',), msg=''):
         if isinstance(suffix, str):
             suffix = [suffix]
         for f in file if isinstance(file, (list, tuple)) else [file]:
-            s = Path(f).suffix.lower()
-            assert (s in suffix) or (len(s) == 0), f"{msg}{f} acceptable suffix is {suffix}"
+            s = Path(f).suffix.lower()  # file suffix
+            if len(s):
+                assert s in suffix, f"{msg}{f} acceptable suffix is {suffix}"
 
 
 def check_yaml(file, suffix=('.yaml', '.yml')):

--- a/utils/general.py
+++ b/utils/general.py
@@ -293,12 +293,13 @@ def check_imshow():
 
 
 def check_suffix(file='yolov5s.pt', suffix=('.pt',), msg=''):
-    # Check file(s) for acceptable suffixes
+    # Check file(s) for acceptable suffix
     if file and suffix:
         if isinstance(suffix, str):
             suffix = [suffix]
         for f in file if isinstance(file, (list, tuple)) else [file]:
-            assert Path(f).suffix.lower() in suffix, f"{msg}{f} acceptable suffix is {suffix}"
+            s = Path(f).suffix.lower()
+            assert (s in suffix) or (len(s) == 0), f"{msg}{f} acceptable suffix is {suffix}"
 
 
 def check_yaml(file, suffix=('.yaml', '.yml')):

